### PR TITLE
chore: OPTIC-1060: Hot Module Reload support for frontend development

### PR DESCRIPTION
**Pull request by bmartel**
### PR fulfills these requirements
- [X] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [X] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [X] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [X] Frontend

[Kapture 2024-08-29 at 17.19.34.webm](https://github.com/user-attachments/assets/d00e4eae-7811-4ab6-ab49-f585e6ccfce3)

### Describe the reason for change
Now that the builds have evolved towards a single entrypoint, single React application with dynamic module imports, we can now easily reap the benefits and implement Hot Module Reload support with very little effort. This has a MASSIVE impact to productivity and contribution. It offers a singular method to develop with, and it vastly reduces cognitive burden by shortening the time to see change to milliseconds, and it is done in realtime to not throw away the context of what you are debugging or working on at that moment.

### Which logical domain(s) does this change affect?
Devtooling/Webpack.

### CODE REVIEW

File: web/webpack.config.js [hunk 1]
```diff
@@ -4,7 +4,10 @@ const { composePlugins, withNx } = require("@nx/webpack");
 const { withReact } = require("@nx/react");
 const { merge } = require("webpack-merge");
 
-require("dotenv").config();
+require("dotenv").config({
+  // resolve the .env file in the root of the project ../
+  path: path.resolve(__dirname, "../.env"),
```
Comment (yyassi-heartex) [inline web/webpack.config.js:9]: what happens if .env folder doesn't exist??

Comment (bmartel) [reply web/webpack.config.js:9]: If the file doesn't exist, it just doesn't read it.

**Review by yyassi-heartex — APPROVED** [review]

**Review by wesleylima — APPROVED** [review]

**Comment by tienldbnk** [comment]
Error: Must provide a proper URL as target when start localhost 8081

**Comment by tienldbnk** [comment]
> ### PR fulfills these requirements
> * [x]  Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
> * [ ]  Tests for the changes have been added/updated (for bug fixes/features)
> * [ ]  Docs have been added/updated (for bug fixes/features)
> * [x]  Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
> * [x]  Self-reviewed and ran all changes on a local instance (for bug fixes/features)
> 
> #### Change has impacts in these area(s)
> _(check all that apply)_
> 
> * [ ]  Product design
> * [ ]  Backend (Database)
> * [ ]  Backend (API)
> * [x]  Frontend
> 
>  Kapture.2024-08-29.at.17.19.34.webm 
> ### Describe the reason for change
> Now that the builds have evolved towards a single entrypoint, single React application with dynamic module imports, we can now easily reap the benefits and implement Hot Module Reload support with very little effort. This has a MASSIVE impact to productivity and contribution. It offers a singular method to develop with, and it vastly reduces cognitive burden by shortening the time to see change to milliseconds, and it is done in realtime to not throw away the context of what you are debugging or working on at that moment.
> 
> ### Which logical domain(s) does this change affect?
> Devtooling/Webpack.

Error: Must provide a proper URL as target when start localhost 8081

**Comment by Bradly-s** [comment]
> > ### PR fulfills these requirements
> > 
> > * [x]   Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
> > * [ ]   Tests for the changes have been added/updated (for bug fixes/features)
> > * [ ]   Docs have been added/updated (for bug fixes/features)
> > * [x]   Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
> > * [x]   Self-reviewed and ran all changes on a local instance (for bug fixes/features)
> > 
> > #### Change has impacts in these area(s)
> > _(check all that apply)_
> > 
> > * [ ]   Product design
> > * [ ]   Backend (Database)
> > * [ ]   Backend (API)
> > * [x]   Frontend
> > 
> > Kapture.2024-08-29.at.17.19.34.webm
> > ### Describe the reason for change
> > Now that the builds have evolved towards a single entrypoint, single React application with dynamic module imports, we can now easily reap the benefits and implement Hot Module Reload support with very little effort. This has a MASSIVE impact to productivity and contribution. It offers a singular method to develop with, and it vastly reduces cognitive burden by shortening the time to see change to milliseconds, and it is done in realtime to not throw away the context of what you are debugging or working on at that moment.
> > ### Which logical domain(s) does this change affect?
> > Devtooling/Webpack.
> 
> Error: Must provide a proper URL as target when start localhost 8081

how to solve it?

**Pull request merged**